automation/taskcluster/decision_task.py

```diff
@@ -26,239 +26,130 @@
-
-    build_task_id = taskcluster.slugId()
-    build_tasks[build_task_id] = BUILDER.craft_assemble_release_task(variant, channel, is_staging, version_name)
-
-    signing_task_id = taskcluster.slugId()
-    signing_tasks[signing_task_id] = BUILDER.craft_release_signing_task(
-        build_task_id,
+    build_task = builder.craft_assemble_release_task(variant, channel, is_staging, version_name)
+    signing_task = builder.craft_release_signing_task(
+        build_task['label'],
         variant.upstream_artifacts(),
         channel,
+        variant,
         is_staging,
     )
 
-    return (build_tasks, signing_tasks)
+    return [build_task, signing_task]
 
 
-def nightly_to_production_app(is_staging, version_name):
+def nightly_to_production_app(builder, is_staging, version_name):
     # Since the Fenix nightly was launched, we've pushed it to the production app "org.mozilla.fenix" on the
     # "nightly" track. We're moving towards having each channel be published to its own app, but we need to
     # keep updating this "backwards-compatible" nightly for a while yet
     variant = get_variant('fenixNightlyLegacy', 'geckoNightly')
     taskcluster_apk_paths = variant.upstream_artifacts()
 
-    build_tasks = {}
-    signing_tasks = {}
-    push_tasks = {}
-    other_tasks = {}
-
-    build_task_id = taskcluster.slugId()
-    build_tasks[build_task_id] = BUILDER.craft_assemble_release_task(
+    build_task = builder.craft_assemble_release_task(
         variant, 'nightly-legacy', is_staging, version_name)
 
-    signing_task_id = taskcluster.slugId()
-    signing_tasks[signing_task_id] = BUILDER.craft_release_signing_task(
-        build_task_id,
+    signing_task = builder.craft_release_signing_task(
+        build_task['label'],
         taskcluster_apk_paths,
         channel='production',  # Since we're publishing to the "production" app, we need to sign for production
         is_staging=is_staging,
         publish_to_index=False,
     )
 
-    push_task_id = taskcluster.slugId()
-    push_tasks[push_task_id] = BUILDER.craft_push_task(
-        signing_task_id,
+    push_task = builder.craft_push_task(
+        signing_task['label'],
         taskcluster_apk_paths,
         channel='production',  # We're publishing to the "production" app on the "nightly" track
+        variant=variant,
         override_google_play_track='nightly',
         is_staging=is_staging,
     )
 
+    tasks = [build_task, signing_task, push_task]
     if not is_staging:
-        nimbledroid_task_id = taskcluster.slugId()
-        other_tasks[nimbledroid_task_id] = BUILDER.craft_upload_apk_nimbledroid_task(
-            build_task_id
-        )
-
-    return (build_tasks, signing_tasks, push_tasks, other_tasks)
-
-
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser(
-        description='Creates and submit a graph of tasks on Taskcluster.'
-    )
+        tasks.append(builder.craft_upload_apk_nimbledroid_task(build_task['label']))
```

Comment:
Sounds like `task['label']` consistently replaces referencing tasks by `task_id` :+1:

Reply:
That's right! The task label is then replaced with a taskId by taskgraph, during the optimization phase. As a consequence, the label must be unique among the graph. That's why some tasks have their name slightly adjusted.